client/me/purchases/purchase-item/index.jsx

```diff
@@ -55,65 +49,95 @@ class PurchaseItem extends Component {
 		);
 	}
 
-	renewsOrExpiresOn() {
+	getStatus() {
 		const { purchase, translate, moment } = this.props;
+		const expiry = moment( purchase.expiryDate );
 
-		if ( showCreditCardExpiringWarning( purchase ) ) {
-			return (
-				<Notice isCompact status="is-error" icon="notice">
-					{ translate( 'Credit card expiring soon' ) }
-					{ this.trackImpression( 'credit-card-expiring' ) }
-				</Notice>
-			);
+		if ( purchase && isPartnerPurchase( purchase ) ) {
+			return translate( 'Managed by %(partnerName)s', {
+				args: {
+					partnerName: getPartnerName( purchase ),
+				},
+			} );
 		}
 
 		if ( isRenewing( purchase ) && purchase.renewDate ) {
 			const renewDate = moment( purchase.renewDate );
 
-			return translate( 'Renews at %(amount)s on %(date)s.', {
+			if ( creditCardHasAlreadyExpired( purchase ) ) {
+				return (
+					<span className="purchase-item__is-error">
+						{ translate( 'Credit card expired' ) }
+						{ this.trackImpression( 'credit-card-expiring' ) }
+					</span>
+				);
+			}
+
+			if ( creditCardExpiresBeforeSubscription( purchase ) ) {
+				return (
+					<span className="purchase-item__is-error">
+						{ translate( 'Credit card expiring soon' ) }
+						{ this.trackImpression( 'credit-card-expiring' ) }
+					</span>
+				);
+			}
+
+			return translate( 'Renews at %(amount)s on {{span}}%(date)s{{/span}}', {
 				args: {
 					amount: purchase.priceText,
 					date: renewDate.format( 'LL' ),
 				},
+				components: {
+					span: <span className="purchase-item__date" />,
+				},
 			} );
 		}
 
-		const expiry = moment( purchase.expiryDate );
-
 		if ( isExpiring( purchase ) ) {
-			if ( expiry < moment().add( 30, 'days' ) ) {
-				const status = isRecentMonthlyPurchase( purchase ) ? 'is-info' : 'is-error';
+			if ( expiry < moment().add( 30, 'days' ) && ! isRecentMonthlyPurchase( purchase ) ) {
+				const expiryClass =
+					expiry < moment().add( 7, 'days' )
+						? 'purchase-item__is-error'
+						: 'purchase-item__is-warning';
+
 				return (
-					<Notice isCompact status={ status } icon="notice">
-						{ translate( 'Expires %(timeUntilExpiry)s', {
+					<span className={ expiryClass }>
+						{ translate( 'Expires %(timeUntilExpiry)s on {{span}}%(date)s{{/span}}', {
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?